File: frontend/src/types/Pet.ts
import Cliente from './Cliente';

export type Pet = {
    id: number
    nome: string
    especie: string
    raca: string
    castrado: boolean
    sexo: string
    dono?: Cliente
}

export type PetInsert = {
    nome: string
    especie: string
    raca: string
    castrado: boolean
    sexo: string
    donoId?: number
}

export default Pet

File: frontweb/Data/Entities/Pet.cs
namespace PetWorldWeb.Data.Entities
{
    public class Pet
    {
        public long Id { get; set; }
        public string Nome { get; set; }
        public string Especie { get; set; }
        public string Raca { get; set; }
        public bool Castrado { get; set; }
        public string Sexo { get; set; }

        public Cliente Dono { get; set; }
    }
}